fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/impl/FHIRPersistenceJDBCImpl.java

```diff
@@ -278,7 +283,7 @@ protected Action buildActionChain() {
             // system-generated value. For the update-or-create scenario, see update().
             // Default version is 1 for a brand new FHIR Resource.
             int newVersionNumber = 1;
-            logicalId = UUID.randomUUID().toString();
+            logicalId = logicalIdentityProvider.createNewIdentityValue();
             if (log.isLoggable(Level.FINE)) {
                 log.fine("Creating new FHIR Resource of type '" + resource.getClass().getSimpleName() + "'");
```

Comment:
For ResourceDAOImpl, Also what about line 518? `stmt.setString(6, UUID.randomUUID().toString());`

Reply:
Also DerbyResourceDao on 117?

Reply:
Yikes...no idea why we'd be manipulating data values in the DAO...good catch...will check

Reply:
That's the sourceKey which was previously used to ensure idempotence in replication scenarios. It is no longer used and can be removed. Will create a new issue. It is benign here.